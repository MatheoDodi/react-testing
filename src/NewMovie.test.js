import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import NewMovie from './NewMovie';

afterEach(cleanup);

test('<NewMovie />', () => {
  const { debug, getByTestId, queryByTestId, container, getByText } = render(
    <NewMovie />
  );
  const newMovie = getByTestId('new-movie');

  expect(newMovie.textContent).toBe('New Movie');
  expect(queryByTestId('movie-form')).toBeTruthy();

  // Creates Snapshot and expects it to always stay the same
  expect(container.firstChild).toMatchSnapshot();
});
