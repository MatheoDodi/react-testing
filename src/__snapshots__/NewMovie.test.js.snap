// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie /> 1`] = `
<div>
  <h2
    data-testid="new-movie"
  >
    New Movie
  </h2>
  <form
    data-testid="movie-form"
  >
    <input
      type="text"
    />
    <button>
      Submit
    </button>
  </form>
</div>
`;
